docs/migrating-v1alpha1.Run-to-v1beta1.CustomRun.md

```diff
@@ -0,0 +1,194 @@
+  - [Timeout](#timeout)
+  - [Retries & RetriesStatus](#4-retries--retriesstatus)
+- [New feature flag `custom-task-version` for migration](#new-feature-flag-custom-task-version)
+
+## Changes to fields
+
+Comparing `v1alpha1.Run` with `v1beta1.CustomRun`, the following fields have been changed:
+
+| Old field              | New field         |
+| ---------------------- | ------------------|
+| `spec.spec`            | [`spec.customSpec`](#changes-to-the-specification) |
+| `spec.ref`             | [`spec.customRef`](#changes-to-the-reference) |
+| `spec.podTemplate`     | removed, see [this section](#support-podtemplate-in-custom-task-spec) if you still want to support it|
+
+
+### Changes to the specification
+
+In `v1beta1.CustomRun`, the specification is renamed from `spec.spec` to `spec.customSpec`.
+
+```yaml
+# before (v1alpha1.Run)
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  name: run-with-spec
+spec:
+  spec:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+    spec:
+      field1: value1
+      field2: value2
+# after (v1beta1.CustomRun)
+apiVersion: tekton.dev/v1beta1
+kind: CustomRun
+metadata:
+  name: customrun-with-spec
+spec:
+  customSpec:
+    apiVersion: example.dev/v1beta1
+    kind: Example
+    spec:
+      field1: value1
+      field2: value2
+```
+
+### Changes to the reference
+
+In `v1beta1.CustomRun`, the specification is renamed from `spec.ref` to `spec.customRef`.
+
+```yaml
+# before (v1alpha1.Run)
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  name: run-with-reference
+spec:
+  ref:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+    name: my-example
+---
+# after (v1beta1.CustomRun)
+apiVersion: tekton.dev/v1beta1
+kind: CustomRun
+metadata:
+  name: customrun-with-reference
+spec:
+  customRef:
+    apiVersion: example.dev/v1beta1
+    kind: Example
+    name: my-customtask
+```
+
+### Support `PodTemplate` in Custom Task Spec
+
+`spec.podTemplate` is removed in `v1beta1.CustomRun`. You can support that field in your own custom task spec if you want to. For example:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: CustomRun
+metadata:
+  name: customrun-with-podtemplate
+spec:
+  customSpec:
+    apiVersion: example.dev/v1beta1
+    kind: Example
+    spec:
+      podTemplate: 
+        securityContext:
+        runAsUser: 1001
+```
+
+## Changes to implementation instructions
+We've changed four implementation instructions. Note that `status reporting` is the only required instruction to follow, others are recommendations.
+
+### Status Reporting
+
+You **MUST** report `v1beta1.CustomRun` as `Done` (set its status as `True` or `False`) **ONLY** when you want Pipeline Controller consider it as finished.
+
+For example, if the `CustomRun` failed, while it still has remaining `retries`. If you want pipeline controller to continue watching its status changing, you **MUST NO** mark it as `Done`. Otherwise, the `PipelineRun` controller may consider it as finished.
```

Comment:
```suggestion
For example, if the `CustomRun` failed, while it still has remaining `retries`. If you want pipeline controller to continue watching its status changing, you **MUST NOT** mark it as `Done`. Otherwise, the `PipelineRun` controller may consider it as finished.
```

Reply:
Done.